eva/optimizer/column_deriver.py

```diff
@@ -0,0 +1,119 @@
+
+#get the optimal plan
+#go to the children attribute
+#types seqscan, project, predicate, lateral join, functional scan
+
+#target list has the list of columns that need to be selected
+
+#Lateral Join: Children traverse sequential and join.
+#Seq Scan: 
+
+
+import sys
+from eva.planner import SeqScanPlan, LateralJoinPlan, StoragePlan, FunctionScanPlan, ProjectPlan
+
+
+if sys.version_info >= (3, 8):
+    from functools import singledispatchmethod
+else:
+    # https://stackoverflow.com/questions/24601722/how-can-i-use-functools-singledispatch-with-instance-methods
+    from functools import singledispatch, update_wrapper
+
+    def singledispatchmethod(func):
+        dispatcher = singledispatch(func)
+
+        def wrapper(*args, **kw):
+            return dispatcher.dispatch(args[1].__class__)(*args, **kw)
+
+        wrapper.register = dispatcher.register
+        update_wrapper(wrapper, func)
+        return wrapper
+
+
+
+
+class ColumnDeriver:
```

Comment:
The logic looks good. However, you need to use better function names and variable names. It is hard to understand what is going on. Thanks!